src/mds/MDSRank.cc

```diff
@@ -576,9 +576,9 @@ bool MDSRank::handle_deferrable_message(Message *m)
       // SERVER
     case CEPH_MSG_CLIENT_SESSION:
     case CEPH_MSG_CLIENT_RECONNECT:
-      ALLOW_MESSAGES_FROM(CEPH_ENTITY_TYPE_CLIENT);
       // fall-thru
     case CEPH_MSG_CLIENT_REQUEST:
+      ALLOW_MESSAGES_FROM(CEPH_ENTITY_TYPE_CLIENT);
```

Comment:
I think you're right @jcsp. Running it through the multi-mds test suite should make it pretty obvious if we're wrong, if we want to do a special run.